cloud-reactor/src/main/java/com/sequenceiq/cloudbreak/cloud/handler/StopStartUpscaleStartInstancesHandler.java

```diff
@@ -51,7 +51,6 @@ public void accept(Event<StopStartUpscaleStartInstancesRequest> event) {
 
         int numInstancesToStart = request.getNumInstancesToStart() - request.getStartedInstancesWithServicesNotRunning().size();
         if (numInstancesToStart <= 0) {
-            // TODO CB-15132: This is currently not exercised, since the Flow itself does not send any good information about running instances.
             LOGGER.debug("No nodes to start. Start requested for numInstances={}, Running CM instances with services not running={}",
                     request.getNumInstancesToStart(), request.getStartedInstancesWithServicesNotRunning().size());
             List<CloudInstance> startedInstancesWithServicesNotRunning = request.getStartedInstancesWithServicesNotRunning();
```

Comment:
There's a discrepancy in the way this class behaves.

if (numInstancesToStartComputed) <=0) - the result contains InstanceStatus entries for the recovery candidates.
If, however, there are some recovery candidates, but not enough to reach the desiredCount - the result does not include the InstanceStatus entries for the recovery candidates.

That is going to cause: 
1. Inconsistent logging at the best.
2. Most likely a functionality break, depending on how the result is being processed in subsequent steps.

Reply:
Updated this to not propagate affectedInstanceStatuses if no nodes were explicitly started via cloud-provider (in case of numInstancesToStart <=0), as discussed